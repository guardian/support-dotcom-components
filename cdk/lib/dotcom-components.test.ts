import { App } from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import { DotcomComponents } from './dotcom-components';

describe('The DotcomComponents stack', () => {
	it('matches the snapshot', () => {
		const app = new App();
		const stack = new DotcomComponents(app, 'DotcomComponents', {
			stack: 'support',
			stage: 'TEST',
			domainName: 'dotcom-components-test.support.guardianapis.com',
		});
		const template = Template.fromStack(stack);
		expect(template.toJSON()).toMatchSnapshot();
	});
});
