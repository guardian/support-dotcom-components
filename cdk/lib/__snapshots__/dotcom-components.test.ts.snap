// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Cfn stack matches the snapshot 1`] = `
Object {
  "Mappings": Object {
    "stagemapping": Object {
      "CODE": Object {
        "alarmActionsEnabled": false,
        "domainName": "dotcom-components-code.support.guardianapis.com",
        "maxInstances": 2,
        "minInstances": 1,
      },
      "PROD": Object {
        "alarmActionsEnabled": true,
        "domainName": "dotcom-components.support.guardianapis.com",
        "maxInstances": 18,
        "minInstances": 3,
      },
    },
  },
  "Outputs": Object {
    "LoadBalancerDotcomcomponentsDnsName": Object {
      "Description": "DNS entry for LoadBalancerDotcomcomponents",
      "Value": Object {
        "Fn::GetAtt": Array [
          "LoadBalancerDotcomcomponentsFF5A6F9F",
          "DNSName",
        ],
      },
    },
  },
  "Parameters": Object {
    "AMIDotcomcomponents": Object {
      "Description": "Amazon Machine Image ID for the app dotcom-components. Use this in conjunction with AMIgo to keep AMIs up to date.",
      "Type": "AWS::EC2::Image::Id",
    },
    "BaseUrl": Object {
      "Description": "Base URL of the service.",
      "Type": "String",
    },
    "DistributionBucketName": Object {
      "Default": "/account/services/artifact.bucket",
      "Description": "SSM parameter containing the S3 bucket name holding distribution artifacts",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
    "ELKStream": Object {
      "Description": "Name of the Kinesis stream used to send logs to the central ELK stack.",
      "Type": "String",
    },
    "InstanceTypeDotcomcomponents": Object {
      "Default": "t3.small",
      "Description": "EC2 Instance Type for the app dotcom-components",
      "Type": "String",
    },
    "LoggingStreamName": Object {
      "Default": "/account/services/logging.stream.name",
      "Description": "SSM parameter containing the Name (not ARN) on the kinesis stream",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
    "Stage": Object {
      "AllowedValues": Array [
        "CODE",
        "PROD",
      ],
      "Default": "CODE",
      "Description": "Stage name",
      "Type": "String",
    },
    "VpcId": Object {
      "Default": "/account/vpc/primary/id",
      "Description": "Virtual Private Cloud to run EC2 instances within",
      "Type": "AWS::SSM::Parameter::Value<AWS::EC2::VPC::Id>",
    },
    "dotcomcomponentsPrivateSubnets": Object {
      "Default": "/account/vpc/primary/subnets/private",
      "Description": "A list of private subnets",
      "Type": "AWS::SSM::Parameter::Value<List<AWS::EC2::Subnet::Id>>",
    },
    "dotcomcomponentsPublicSubnets": Object {
      "Default": "/account/vpc/primary/subnets/public",
      "Description": "A list of public subnets",
      "Type": "AWS::SSM::Parameter::Value<List<AWS::EC2::Subnet::Id>>",
    },
  },
  "Resources": Object {
    "AutoScalingGroupDotcomcomponentsASG512C0EA8": Object {
      "Properties": Object {
        "HealthCheckGracePeriod": 120,
        "HealthCheckType": "ELB",
        "LaunchConfigurationName": Object {
          "Ref": "AutoScalingGroupDotcomcomponentsLaunchConfig85091051",
        },
        "MaxSize": Object {
          "Fn::FindInMap": Array [
            "stagemapping",
            Object {
              "Ref": "Stage",
            },
            "maxInstances",
          ],
        },
        "MinSize": Object {
          "Fn::FindInMap": Array [
            "stagemapping",
            Object {
              "Ref": "Stage",
            },
            "minInstances",
          ],
        },
        "Tags": Array [
          Object {
            "Key": "App",
            "PropagateAtLaunch": true,
            "Value": "dotcom-components",
          },
          Object {
            "Key": "gu:cdk:pattern-name",
            "PropagateAtLaunch": true,
            "Value": "GuEc2App",
          },
          Object {
            "Key": "gu:cdk:version",
            "PropagateAtLaunch": true,
            "Value": "23.5.1",
          },
          Object {
            "Key": "gu:repo",
            "PropagateAtLaunch": true,
            "Value": "guardian/support-dotcom-components",
          },
          Object {
            "Key": "Name",
            "PropagateAtLaunch": true,
            "Value": "cfn/AutoScalingGroupDotcomcomponents",
          },
          Object {
            "Key": "Stack",
            "PropagateAtLaunch": true,
            "Value": "support",
          },
          Object {
            "Key": "Stage",
            "PropagateAtLaunch": true,
            "Value": Object {
              "Ref": "Stage",
            },
          },
        ],
        "TargetGroupARNs": Array [
          Object {
            "Ref": "TargetGroupDotcomcomponents54FF73AB",
          },
        ],
        "VPCZoneIdentifier": Object {
          "Ref": "dotcomcomponentsPrivateSubnets",
        },
      },
      "Type": "AWS::AutoScaling::AutoScalingGroup",
    },
    "AutoScalingGroupDotcomcomponentsInstanceProfile55946DEC": Object {
      "Properties": Object {
        "Roles": Array [
          Object {
            "Ref": "InstanceRoleDotcomcomponents2E8FDE7D",
          },
        ],
      },
      "Type": "AWS::IAM::InstanceProfile",
    },
    "AutoScalingGroupDotcomcomponentsLaunchConfig85091051": Object {
      "DependsOn": Array [
        "InstanceRoleDotcomcomponents2E8FDE7D",
      ],
      "Properties": Object {
        "IamInstanceProfile": Object {
          "Ref": "AutoScalingGroupDotcomcomponentsInstanceProfile55946DEC",
        },
        "ImageId": Object {
          "Ref": "AMIDotcomcomponents",
        },
        "InstanceType": Object {
          "Ref": "InstanceTypeDotcomcomponents",
        },
        "SecurityGroups": Array [
          Object {
            "Fn::GetAtt": Array [
              "GuHttpsEgressSecurityGroupDotcomcomponents3638A075",
              "GroupId",
            ],
          },
          Object {
            "Fn::GetAtt": Array [
              "WazuhSecurityGroup",
              "GroupId",
            ],
          },
        ],
        "UserData": Object {
          "Fn::Base64": Object {
            "Fn::Join": Array [
              "",
              Array [
                "#!/bin/bash

groupadd support
useradd -r -m -s /usr/bin/nologin -g support dotcom-components
cd /home/dotcom-components

aws --region eu-west-1 s3 cp s3://",
                Object {
                  "Ref": "DistributionBucketName",
                },
                "/support/",
                Object {
                  "Ref": "Stage",
                },
                "/dotcom-components/dotcom-components.tar ./
mkdir dotcom-components
tar -xvf dotcom-components.tar --directory dotcom-components

chown -R dotcom-components:support dotcom-components

cd dotcom-components

export TERM=xterm-256color
export NODE_ENV=production
export stage=",
                Object {
                  "Ref": "Stage",
                },
                "
export base_url=",
                Object {
                  "Ref": "BaseUrl",
                },
                "

mkdir /var/log/dotcom-components
chown -R dotcom-components:support /var/log/dotcom-components

/usr/local/node/pm2 start --uid dotcom-components --gid support server.js --max-memory-restart 3000M

/opt/aws-kinesis-agent/configure-aws-kinesis-agent ",
                Object {
                  "Ref": "AWS::Region",
                },
                " ",
                Object {
                  "Ref": "ELKStream",
                },
                " /var/log/dotcom-components/dotcom-components.log",
              ],
            ],
          },
        },
      },
      "Type": "AWS::AutoScaling::LaunchConfiguration",
    },
    "AutoScalingGroupDotcomcomponentsScalingPolicyCpuScalingPolicy29FF9AE6": Object {
      "Properties": Object {
        "AutoScalingGroupName": Object {
          "Ref": "AutoScalingGroupDotcomcomponentsASG512C0EA8",
        },
        "PolicyType": "TargetTrackingScaling",
        "TargetTrackingConfiguration": Object {
          "PredefinedMetricSpecification": Object {
            "PredefinedMetricType": "ASGAverageCPUUtilization",
          },
          "TargetValue": 50,
        },
      },
      "Type": "AWS::AutoScaling::ScalingPolicy",
    },
    "CertificateDotcomcomponents88C2E1C7": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "DomainName": Object {
          "Fn::FindInMap": Array [
            "stagemapping",
            Object {
              "Ref": "Stage",
            },
            "domainName",
          ],
        },
        "Tags": Array [
          Object {
            "Key": "App",
            "Value": "dotcom-components",
          },
          Object {
            "Key": "gu:cdk:version",
            "Value": "23.5.1",
          },
          Object {
            "Key": "gu:repo",
            "Value": "guardian/support-dotcom-components",
          },
          Object {
            "Key": "Stack",
            "Value": "support",
          },
          Object {
            "Key": "Stage",
            "Value": Object {
              "Ref": "Stage",
            },
          },
        ],
        "ValidationMethod": "DNS",
      },
      "Type": "AWS::CertificateManager::Certificate",
      "UpdateReplacePolicy": "Retain",
    },
    "DescribeEC2PolicyFF5F9295": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "autoscaling:DescribeAutoScalingInstances",
                "autoscaling:DescribeAutoScalingGroups",
                "ec2:DescribeTags",
                "ec2:DescribeInstances",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "describe-ec2-policy",
        "Roles": Array [
          Object {
            "Ref": "InstanceRoleDotcomcomponents2E8FDE7D",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "DynamoReadPolicy15DC9FD5": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "dynamodb:BatchGetItem",
                "dynamodb:GetItem",
                "dynamodb:Scan",
                "dynamodb:Query",
                "dynamodb:GetRecords",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    "arn:aws:dynamodb:",
                    Object {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    Object {
                      "Ref": "AWS::AccountId",
                    },
                    ":table/super-mode-",
                    Object {
                      "Ref": "Stage",
                    },
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "DynamoReadPolicy15DC9FD5",
        "Roles": Array [
          Object {
            "Ref": "InstanceRoleDotcomcomponents2E8FDE7D",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "DynamoReadPolicySecondaryIndex1A0FDE89": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "dynamodb:BatchGetItem",
                "dynamodb:GetItem",
                "dynamodb:Scan",
                "dynamodb:Query",
                "dynamodb:GetRecords",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    "arn:aws:dynamodb:",
                    Object {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    Object {
                      "Ref": "AWS::AccountId",
                    },
                    ":table/super-mode-",
                    Object {
                      "Ref": "Stage",
                    },
                    "/index/*",
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "DynamoReadPolicySecondaryIndex1A0FDE89",
        "Roles": Array [
          Object {
            "Ref": "InstanceRoleDotcomcomponents2E8FDE7D",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "GetDistributablePolicyDotcomcomponentsDB5B987B": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "s3:GetObject",
              "Effect": "Allow",
              "Resource": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    "arn:aws:s3:::",
                    Object {
                      "Ref": "DistributionBucketName",
                    },
                    "/support/",
                    Object {
                      "Ref": "Stage",
                    },
                    "/dotcom-components/*",
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "GetDistributablePolicyDotcomcomponentsDB5B987B",
        "Roles": Array [
          Object {
            "Ref": "InstanceRoleDotcomcomponents2E8FDE7D",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "GuHttpsEgressSecurityGroupDotcomcomponents3638A075": Object {
      "Properties": Object {
        "GroupDescription": "Allow all outbound HTTPS traffic",
        "SecurityGroupEgress": Array [
          Object {
            "CidrIp": "0.0.0.0/0",
            "Description": "Allow all outbound HTTPS traffic",
            "FromPort": 443,
            "IpProtocol": "tcp",
            "ToPort": 443,
          },
        ],
        "Tags": Array [
          Object {
            "Key": "App",
            "Value": "dotcom-components",
          },
          Object {
            "Key": "gu:cdk:version",
            "Value": "23.5.1",
          },
          Object {
            "Key": "gu:repo",
            "Value": "guardian/support-dotcom-components",
          },
          Object {
            "Key": "Stack",
            "Value": "support",
          },
          Object {
            "Key": "Stage",
            "Value": Object {
              "Ref": "Stage",
            },
          },
        ],
        "VpcId": Object {
          "Ref": "VpcId",
        },
      },
      "Type": "AWS::EC2::SecurityGroup",
    },
    "GuHttpsEgressSecurityGroupDotcomcomponentsfromcfnLoadBalancerDotcomcomponentsSecurityGroupE953877E3030690B8132": Object {
      "Properties": Object {
        "Description": "Load balancer to target",
        "FromPort": 3030,
        "GroupId": Object {
          "Fn::GetAtt": Array [
            "GuHttpsEgressSecurityGroupDotcomcomponents3638A075",
            "GroupId",
          ],
        },
        "IpProtocol": "tcp",
        "SourceSecurityGroupId": Object {
          "Fn::GetAtt": Array [
            "LoadBalancerDotcomcomponentsSecurityGroupD06C9706",
            "GroupId",
          ],
        },
        "ToPort": 3030,
      },
      "Type": "AWS::EC2::SecurityGroupIngress",
    },
    "GuLogShippingPolicy981BFE5A": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "kinesis:Describe*",
                "kinesis:Put*",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    "arn:aws:kinesis:",
                    Object {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    Object {
                      "Ref": "AWS::AccountId",
                    },
                    ":stream/",
                    Object {
                      "Ref": "LoggingStreamName",
                    },
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "GuLogShippingPolicy981BFE5A",
        "Roles": Array [
          Object {
            "Ref": "InstanceRoleDotcomcomponents2E8FDE7D",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "High5xxPercentageAlarmDotcomcomponents70DE8F71": Object {
      "Properties": Object {
        "ActionsEnabled": Object {
          "Fn::FindInMap": Array [
            "stagemapping",
            Object {
              "Ref": "Stage",
            },
            "alarmActionsEnabled",
          ],
        },
        "AlarmActions": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:aws:sns:",
                Object {
                  "Ref": "AWS::Region",
                },
                ":",
                Object {
                  "Ref": "AWS::AccountId",
                },
                ":reader-revenue-dev",
              ],
            ],
          },
        ],
        "AlarmDescription": "dotcom-components exceeded 0.1% error rate",
        "AlarmName": Object {
          "Fn::Join": Array [
            "",
            Array [
              "URGENT 9-5 - high 5XX error rate on ",
              Object {
                "Ref": "Stage",
              },
              " support-dotcom-components",
            ],
          ],
        },
        "ComparisonOperator": "GreaterThanThreshold",
        "EvaluationPeriods": 1,
        "Metrics": Array [
          Object {
            "Expression": "100*(m1+m2)/m3",
            "Id": "expr_1",
            "Label": "% of 5XX responses served for dotcom-components (load balancer and instances combined)",
          },
          Object {
            "Id": "m1",
            "MetricStat": Object {
              "Metric": Object {
                "Dimensions": Array [
                  Object {
                    "Name": "LoadBalancer",
                    "Value": Object {
                      "Fn::GetAtt": Array [
                        "LoadBalancerDotcomcomponentsFF5A6F9F",
                        "LoadBalancerFullName",
                      ],
                    },
                  },
                ],
                "MetricName": "HTTPCode_ELB_5XX_Count",
                "Namespace": "AWS/ApplicationELB",
              },
              "Period": 60,
              "Stat": "Sum",
            },
            "ReturnData": false,
          },
          Object {
            "Id": "m2",
            "MetricStat": Object {
              "Metric": Object {
                "Dimensions": Array [
                  Object {
                    "Name": "LoadBalancer",
                    "Value": Object {
                      "Fn::GetAtt": Array [
                        "LoadBalancerDotcomcomponentsFF5A6F9F",
                        "LoadBalancerFullName",
                      ],
                    },
                  },
                ],
                "MetricName": "HTTPCode_Target_5XX_Count",
                "Namespace": "AWS/ApplicationELB",
              },
              "Period": 60,
              "Stat": "Sum",
            },
            "ReturnData": false,
          },
          Object {
            "Id": "m3",
            "MetricStat": Object {
              "Metric": Object {
                "Dimensions": Array [
                  Object {
                    "Name": "LoadBalancer",
                    "Value": Object {
                      "Fn::GetAtt": Array [
                        "LoadBalancerDotcomcomponentsFF5A6F9F",
                        "LoadBalancerFullName",
                      ],
                    },
                  },
                ],
                "MetricName": "RequestCount",
                "Namespace": "AWS/ApplicationELB",
              },
              "Period": 60,
              "Stat": "Sum",
            },
            "ReturnData": false,
          },
        ],
        "Threshold": 0.1,
        "TreatMissingData": "notBreaching",
      },
      "Type": "AWS::CloudWatch::Alarm",
    },
    "InstanceRoleDotcomcomponents2E8FDE7D": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "ec2.",
                      Object {
                        "Ref": "AWS::URLSuffix",
                      },
                    ],
                  ],
                },
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "Path": "/",
        "Tags": Array [
          Object {
            "Key": "App",
            "Value": "dotcom-components",
          },
          Object {
            "Key": "gu:cdk:version",
            "Value": "23.5.1",
          },
          Object {
            "Key": "gu:repo",
            "Value": "guardian/support-dotcom-components",
          },
          Object {
            "Key": "Stack",
            "Value": "support",
          },
          Object {
            "Key": "Stage",
            "Value": Object {
              "Ref": "Stage",
            },
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "ListenerDotcomcomponentsF68CDC56": Object {
      "Properties": Object {
        "Certificates": Array [
          Object {
            "CertificateArn": Object {
              "Ref": "CertificateDotcomcomponents88C2E1C7",
            },
          },
        ],
        "DefaultActions": Array [
          Object {
            "TargetGroupArn": Object {
              "Ref": "TargetGroupDotcomcomponents54FF73AB",
            },
            "Type": "forward",
          },
        ],
        "LoadBalancerArn": Object {
          "Ref": "LoadBalancerDotcomcomponentsFF5A6F9F",
        },
        "Port": 443,
        "Protocol": "HTTPS",
      },
      "Type": "AWS::ElasticLoadBalancingV2::Listener",
    },
    "LoadBalancerDotcomcomponentsFF5A6F9F": Object {
      "Properties": Object {
        "LoadBalancerAttributes": Array [
          Object {
            "Key": "deletion_protection.enabled",
            "Value": "true",
          },
        ],
        "Scheme": "internet-facing",
        "SecurityGroups": Array [
          Object {
            "Fn::GetAtt": Array [
              "LoadBalancerDotcomcomponentsSecurityGroupD06C9706",
              "GroupId",
            ],
          },
        ],
        "Subnets": Object {
          "Ref": "dotcomcomponentsPublicSubnets",
        },
        "Tags": Array [
          Object {
            "Key": "App",
            "Value": "dotcom-components",
          },
          Object {
            "Key": "gu:cdk:version",
            "Value": "23.5.1",
          },
          Object {
            "Key": "gu:repo",
            "Value": "guardian/support-dotcom-components",
          },
          Object {
            "Key": "Stack",
            "Value": "support",
          },
          Object {
            "Key": "Stage",
            "Value": Object {
              "Ref": "Stage",
            },
          },
        ],
        "Type": "application",
      },
      "Type": "AWS::ElasticLoadBalancingV2::LoadBalancer",
    },
    "LoadBalancerDotcomcomponentsSecurityGroupD06C9706": Object {
      "Properties": Object {
        "GroupDescription": "Automatically created Security Group for ELB cfnLoadBalancerDotcomcomponentsFFDD63DE",
        "SecurityGroupIngress": Array [
          Object {
            "CidrIp": "0.0.0.0/0",
            "Description": "Allow from anyone on port 443",
            "FromPort": 443,
            "IpProtocol": "tcp",
            "ToPort": 443,
          },
        ],
        "Tags": Array [
          Object {
            "Key": "App",
            "Value": "dotcom-components",
          },
          Object {
            "Key": "gu:cdk:version",
            "Value": "23.5.1",
          },
          Object {
            "Key": "gu:repo",
            "Value": "guardian/support-dotcom-components",
          },
          Object {
            "Key": "Stack",
            "Value": "support",
          },
          Object {
            "Key": "Stage",
            "Value": Object {
              "Ref": "Stage",
            },
          },
        ],
        "VpcId": Object {
          "Ref": "VpcId",
        },
      },
      "Type": "AWS::EC2::SecurityGroup",
    },
    "LoadBalancerDotcomcomponentsSecurityGrouptocfnGuHttpsEgressSecurityGroupDotcomcomponents5B0C5960303092CC03B2": Object {
      "Properties": Object {
        "Description": "Load balancer to target",
        "DestinationSecurityGroupId": Object {
          "Fn::GetAtt": Array [
            "GuHttpsEgressSecurityGroupDotcomcomponents3638A075",
            "GroupId",
          ],
        },
        "FromPort": 3030,
        "GroupId": Object {
          "Fn::GetAtt": Array [
            "LoadBalancerDotcomcomponentsSecurityGroupD06C9706",
            "GroupId",
          ],
        },
        "IpProtocol": "tcp",
        "ToPort": 3030,
      },
      "Type": "AWS::EC2::SecurityGroupEgress",
    },
    "ParameterStoreReadDotcomcomponents7A7C8BD4": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "ssm:GetParametersByPath",
              "Effect": "Allow",
              "Resource": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    "arn:aws:ssm:",
                    Object {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    Object {
                      "Ref": "AWS::AccountId",
                    },
                    ":parameter/",
                    Object {
                      "Ref": "Stage",
                    },
                    "/support/dotcom-components",
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "parameter-store-read-policy",
        "Roles": Array [
          Object {
            "Ref": "InstanceRoleDotcomcomponents2E8FDE7D",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "S3ReadPolicyGuContributionsPublic821CE426": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "s3:GetObject",
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:aws:s3:::gu-contributions-public/epic/",
                      Object {
                        "Ref": "Stage",
                      },
                      "/*",
                    ],
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:aws:s3:::gu-contributions-public/banner/",
                      Object {
                        "Ref": "Stage",
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "S3ReadPolicyGuContributionsPublic821CE426",
        "Roles": Array [
          Object {
            "Ref": "InstanceRoleDotcomcomponents2E8FDE7D",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "S3ReadPolicySupportAdminConsole06C5574B": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "s3:GetObject",
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:aws:s3:::support-admin-console/",
                      Object {
                        "Ref": "Stage",
                      },
                      "/banner-deploy/*",
                    ],
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:aws:s3:::support-admin-console/",
                      Object {
                        "Ref": "Stage",
                      },
                      "/channel-switches.json",
                    ],
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:aws:s3:::support-admin-console/",
                      Object {
                        "Ref": "Stage",
                      },
                      "/configured-amounts.json",
                    ],
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:aws:s3:::support-admin-console/",
                      Object {
                        "Ref": "Stage",
                      },
                      "/single-contributor-propensity-test",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "S3ReadPolicySupportAdminConsole06C5574B",
        "Roles": Array [
          Object {
            "Ref": "InstanceRoleDotcomcomponents2E8FDE7D",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "SSMRunCommandPolicy244E1613": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "ec2messages:AcknowledgeMessage",
                "ec2messages:DeleteMessage",
                "ec2messages:FailMessage",
                "ec2messages:GetEndpoint",
                "ec2messages:GetMessages",
                "ec2messages:SendReply",
                "ssm:UpdateInstanceInformation",
                "ssm:ListInstanceAssociations",
                "ssm:DescribeInstanceProperties",
                "ssm:DescribeDocumentParameters",
                "ssmmessages:CreateControlChannel",
                "ssmmessages:CreateDataChannel",
                "ssmmessages:OpenControlChannel",
                "ssmmessages:OpenDataChannel",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "ssm-run-command-policy",
        "Roles": Array [
          Object {
            "Ref": "InstanceRoleDotcomcomponents2E8FDE7D",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "TargetGroupDotcomcomponents54FF73AB": Object {
      "Properties": Object {
        "HealthCheckIntervalSeconds": 10,
        "HealthCheckPath": "/healthcheck",
        "HealthCheckProtocol": "HTTP",
        "HealthCheckTimeoutSeconds": 5,
        "HealthyThresholdCount": 5,
        "Port": 3030,
        "Protocol": "HTTP",
        "Tags": Array [
          Object {
            "Key": "App",
            "Value": "dotcom-components",
          },
          Object {
            "Key": "gu:cdk:version",
            "Value": "23.5.1",
          },
          Object {
            "Key": "gu:repo",
            "Value": "guardian/support-dotcom-components",
          },
          Object {
            "Key": "Stack",
            "Value": "support",
          },
          Object {
            "Key": "Stage",
            "Value": Object {
              "Ref": "Stage",
            },
          },
        ],
        "TargetGroupAttributes": Array [
          Object {
            "Key": "deregistration_delay.timeout_seconds",
            "Value": "30",
          },
        ],
        "TargetType": "instance",
        "UnhealthyThresholdCount": 2,
        "VpcId": Object {
          "Ref": "VpcId",
        },
      },
      "Type": "AWS::ElasticLoadBalancingV2::TargetGroup",
    },
    "UnhealthyInstancesAlarmDotcomcomponentsB1904A86": Object {
      "Properties": Object {
        "ActionsEnabled": Object {
          "Fn::FindInMap": Array [
            "stagemapping",
            Object {
              "Ref": "Stage",
            },
            "alarmActionsEnabled",
          ],
        },
        "AlarmActions": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:aws:sns:",
                Object {
                  "Ref": "AWS::Region",
                },
                ":",
                Object {
                  "Ref": "AWS::AccountId",
                },
                ":reader-revenue-dev",
              ],
            ],
          },
        ],
        "AlarmDescription": "dotcom-components's instances have failed healthchecks several times over the last hour.
      This typically results in the AutoScaling Group cycling instances and can lead to problems with deployment,
      scaling or handling traffic spikes.

      Check dotcom-components's application logs or ssh onto an unhealthy instance in order to debug these problems.",
        "AlarmName": Object {
          "Fn::Join": Array [
            "",
            Array [
              "Unhealthy instances for dotcom-components in ",
              Object {
                "Ref": "Stage",
              },
            ],
          ],
        },
        "ComparisonOperator": "GreaterThanOrEqualToThreshold",
        "DatapointsToAlarm": 6,
        "Dimensions": Array [
          Object {
            "Name": "LoadBalancer",
            "Value": Object {
              "Fn::Join": Array [
                "",
                Array [
                  Object {
                    "Fn::Select": Array [
                      1,
                      Object {
                        "Fn::Split": Array [
                          "/",
                          Object {
                            "Ref": "ListenerDotcomcomponentsF68CDC56",
                          },
                        ],
                      },
                    ],
                  },
                  "/",
                  Object {
                    "Fn::Select": Array [
                      2,
                      Object {
                        "Fn::Split": Array [
                          "/",
                          Object {
                            "Ref": "ListenerDotcomcomponentsF68CDC56",
                          },
                        ],
                      },
                    ],
                  },
                  "/",
                  Object {
                    "Fn::Select": Array [
                      3,
                      Object {
                        "Fn::Split": Array [
                          "/",
                          Object {
                            "Ref": "ListenerDotcomcomponentsF68CDC56",
                          },
                        ],
                      },
                    ],
                  },
                ],
              ],
            },
          },
          Object {
            "Name": "TargetGroup",
            "Value": Object {
              "Fn::GetAtt": Array [
                "TargetGroupDotcomcomponents54FF73AB",
                "TargetGroupFullName",
              ],
            },
          },
        ],
        "EvaluationPeriods": 12,
        "MetricName": "UnHealthyHostCount",
        "Namespace": "AWS/ApplicationELB",
        "Period": 300,
        "Statistic": "Maximum",
        "Threshold": 1,
        "TreatMissingData": "notBreaching",
      },
      "Type": "AWS::CloudWatch::Alarm",
    },
    "WazuhSecurityGroup": Object {
      "Properties": Object {
        "GroupDescription": "Allow outbound traffic from wazuh agent to manager",
        "SecurityGroupEgress": Array [
          Object {
            "CidrIp": "0.0.0.0/0",
            "Description": "Wazuh event logging",
            "FromPort": 1514,
            "IpProtocol": "tcp",
            "ToPort": 1514,
          },
          Object {
            "CidrIp": "0.0.0.0/0",
            "Description": "Wazuh agent registration",
            "FromPort": 1515,
            "IpProtocol": "tcp",
            "ToPort": 1515,
          },
        ],
        "Tags": Array [
          Object {
            "Key": "App",
            "Value": "dotcom-components",
          },
          Object {
            "Key": "gu:cdk:version",
            "Value": "23.5.1",
          },
          Object {
            "Key": "gu:repo",
            "Value": "guardian/support-dotcom-components",
          },
          Object {
            "Key": "Stack",
            "Value": "support",
          },
          Object {
            "Key": "Stage",
            "Value": Object {
              "Ref": "Stage",
            },
          },
        ],
        "VpcId": Object {
          "Ref": "VpcId",
        },
      },
      "Type": "AWS::EC2::SecurityGroup",
    },
  },
}
`;
